### PR TITLE
cleverer routing of upstream errors to splunk/sentry

### PR DESCRIPTION
**Pull request by wheresrhys**
@matthew-andrews

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,7 +8,8 @@
   },
   "license": "MIT",
   "dependencies": {
-    "ft-next-logger": "^2.2.1",
+    "fetchres": "^1.5.1",
+    "ft-next-logger": "^2.2.3",
```
Comment (ironsidevsquincy) [inline package.json]: could get the last logger too, v3, has a slight api change - probably just need to change to

```
const logger = require('ft-next-logger').logger;
```

Comment (wheresrhys) [inline package.json]: ta

File: main.js [hunk 2]
```diff
@@ -1,12 +1,49 @@
...
 
-const logger = require('ft-next-logger');
+const logger = require('ft-next-logger').logger;
+const fetchres = require('fetchres');
+const raven = require('raven');
+let ravenMiddleware;
+
+function sendErrorDev (err, req, res) {
+	if (err.name === fetchres.ReadTimeoutError.name) {
+		logger.error('Dependency timeout -', err);
```
Comment (matthew-andrews) [inline main.js]: `event=depedencytimeout`? for easier splunk searches

Comment (matthew-andrews) [inline main.js]: `event=depedencytimeout`? for easier splunk searches

Comment (wheresrhys) [inline main.js]: ah yes, I wasn't sure of best error formatting. ft-next-logger should document this

Comment (matthew-andrews) [inline main.js]: yeah we should docuemnt better…… just standard splunk format really…  `property=value`  i like using `event`

Comment (ironsidevsquincy) [inline main.js]: yeah, need to update the logs, but you can pass a error message string in the first argument, and an object in the second that will be turned into splunk fields, e.g.

```
logger.error('Whoops, there\'s been an error', { event: ' depedencytimeout' });
```

Comment (wheresrhys) [inline main.js]: I had a feeling that was the case. Does that work with an arbitrary list of objects e.g. `error('oops', {custom: 'prop'}, err)`

Comment (ironsidevsquincy) [inline main.js]: hmmmm, don't *think* so, worth testing. if not, maybe

```
logger.error('opps', Object.assign({}, { custom: 'prop'}, err));
```

?

Comment (wheresrhys) [inline main.js]: Neat. But feels like that sort of gymnastics is worth building into next logger, which I could do later today if it seems like a good idea

Comment (ironsidevsquincy) [inline main.js]: Absolutely, just check Winston isn't doing it behind the scenes. Wanted to make logger a bit more cleverer around error object generally, as those are the things that are usually passed to it

**Pull request merged**